frontend/src/lib/components/icon/Icon.svelte

```diff
@@ -0,0 +1,108 @@
+<script lang="ts">
+  import {_} from 'svelte-i18n';
+  import type {IconProps} from './Icon.type';
+  import {ICONS} from './icons';
+
+  type $$Props = IconProps;
+  export let name: $$Props['name'];
+  export let size: $$Props['size'] = 'md';
+  export let color: $$Props['color'] = 'current';
+
+  // Validate name, split path and get aria-label
+  // We need this part-wise approach because of Vite's dynamic import limitations
+  // https://github.com/rollup/plugins/tree/master/packages/dynamic-import-vars#limitations
+  let label: string;
+  let folder: string;
+  let filename: string;
+
+  if (name in ICONS) {
+    label = $_(`icons.${name}`);
+    [folder, filename] = ICONS[name].split('/');
+  } else {
+    label = name;
+    folder = 'custom';
+    filename = 'missing_icon';
+  }
+
+  // Create class names
+  let classes = 'vaa-icon inline';
+
+  // Predefined sizes
+  switch (size) {
+    case 'sm':
+      classes += ' h-16 w-16';
+      break;
+    case 'lg':
+      classes += ' h-32 w-32';
+      break;
+    default:
+      classes += ' h-24 w-24';
+  }
+
+  // Fill colors
+  // NB. We need to spell these out because otherwise Tailwind won't compile them
+  const FILLS = {
+    current: 'fill-current',
+    primary: 'fill-primary',
+    secondary: 'fill-secondary',
+    accent: 'fill-accent',
+    neutral: 'fill-neutral',
+    'base-100': 'fill-base-100',
+    'base-200': 'fill-base-200',
+    'base-300': 'fill-base-300',
+    info: 'fill-info',
+    success: 'fill-success',
+    warning: 'fill-warning',
+    error: 'fill-error',
+    'base-content': 'fill-base-content',
+    'primary-content': 'fill-primary-content',
+    'secondary-content': 'fill-secondary-content',
+    'accent-content': 'fill-accent-content',
+    'info-content': 'fill-info-content',
+    'success-content': 'fill-success-content',
+    'warning-content': 'fill-warning-content',
+    'error-content': 'fill-error-content'
+  };
+
+  if (color != null && color in FILLS) {
+    classes += ` ${FILLS[color]}`;
+  }
+
+  // Merge classes into $$restProps
+  $$restProps.class = `${classes} ${$$restProps.class ?? ''}`;
+</script>
+
+<!--
+@component
+An icon component, where the `name` property defines which icon to use.
+
+Use the other properties to set the size and color of the icon. The
+icon will get a default `aria-label` but it's best you supply one that
+describes its function better. You can also pass any valid attributes 
+to the `div` element wrapping the `svg` icon.
+
+### Properties
+
+- See `Icon.type.ts` or code completion info.
+
+### Usage
+
+```tsx
+<Icon name="addToList" color="primary" 
+ aria-label="Add candidate to your list" aria-hidden="true"/>
+```
+-->
+
+<div aria-label={label} {...$$restProps}>
+  {#await import(`./svg/${folder}/${filename}.svg?raw`) then svg}
```

Comment:
Regarding the lazy loading, Why not add the `svg` files directly in the `/static/icons` folder? If we add the icons there, we could benefit from caching.

Reply:
I was checking the network usage, and it seems they are still cached even from where they currently are, if I understand correctly You can try this by jumping between the two pages in `_test` now.